wire-ios-data-model/Source/Model/User/ZMUser+LegalHoldRequest.swift

```diff
@@ -230,21 +230,29 @@ extension ZMUser: SelfLegalHoldSubject {
      * - returns: The created client, if the state is valid.
      */
 
-    public func addLegalHoldClient(from request: LegalHoldRequest) -> UserClient? {
-        guard let moc = self.managedObjectContext, let selfClient = self.selfClient() else { return nil }
+    public func addLegalHoldClient(from request: LegalHoldRequest) async -> UserClient? {
+        guard
+            let context = self.managedObjectContext,
+            let selfClient = self.selfClient(),
```

Comment:
May `selfClient()` be called from an async method without `context.perform`?

Reply:
I would say it needs to be changed yes cc @typfel

Reply:
Fixed